### DISCUSSION TITLE
`Error: direct DDL is disabled`  How do I fix this?

### DISCUSSION THREAD

**Discussion by christopheragnus** [prisma/prisma#16712] [Q&A]
Hi there,

I made a edit to my `schema.prisma` so I can add a new table with new columns.

I connected to my non-main branch using:

`pscale connect nform-dev add-landingpage-table --port 3309`

My branch list looks like this:

```
 NAME                    PARENT BRANCH   PRODUCTION   READY   CREATED AT       UPDATED AT
 ----------------------- --------------- ------------ ------- ---------------- ----------------
  main                    n/a             Yes          Yes     1 week ago       45 minutes ago
  add-landingpage-table   main            No           Yes     34 minutes ago   33 minutes ago
```

But I get an error after running:

`npx prisma db push`

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "nform-dev" at "aws-ap-southeast-2.connect.psdb.cloud:3306"
Error: direct DDL is disabled
   0: sql_migration_connector::apply_migration::migration_step
           with step=CreateTable { table_id: TableId(5) }
             at migration-engine/connectors/sql-migration-connector/src/apply_migration.rs:21
   1: sql_migration_connector::apply_migration::apply_migration
             at migration-engine/connectors/sql-migration-connector/src/apply_migration.rs:10
   2: migration_core::state::SchemaPush
             at migration-engine/core/src/state.rs:381
```

I double-checked I wasn't pushing to a production branch and I'm still getting this error.

**Comment by christopheragnus** [comment]
I have also checked this link and followed the instructions for executing succeeding Prisma migrations: https://planetscale.com/docs/tutorials/automatic-prisma-migrations#execute-succeeding-prisma-migrations-in-planetscale

I have created a new development branch from main.
Closed the proxy connection to my `main` branch and connected it to `add-landingpage-table` branch.


My Prisma Schema file is adding a new model call LandingPageContent: 

```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

model User {
  id            String    @id @default(cuid())
  name          String?
  // if you are using Github OAuth, you can get rid of the username attribute (that is for Twitter OAuth)
  username      String?
  gh_username   String?
  email         String?   @unique
  emailVerified DateTime?
  image         String?
  createdAt     DateTime  @default(now())
  updatedAt     DateTime  @updatedAt
  accounts      Account[]
  sessions      Session[]
  sites         Site[]
}

model Account {
  id                       String  @id @default(cuid())
  userId                   String
  type                     String
  provider                 String
  providerAccountId        String
  refresh_token            String?
  refresh_token_expires_in Int?
  access_token             String?
  expires_at               Int?
  token_type               String?
  scope                    String?
  id_token                 String?
  session_state            String?
  oauth_token_secret       String?
  oauth_token              String?

  user User @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
  @@index([userId])
}

model Session {
  id           String   @id @default(cuid())
  sessionToken String   @unique
  userId       String
  expires      DateTime
  user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@index([userId])
}

model VerificationToken {
  identifier String
  token      String   @unique
  expires    DateTime

  @@unique([identifier, token])
}

model Post {
  id            String   @id @default(cuid())
  title         String?  @db.Text
  description   String?  @db.Text
  content       String?  @db.LongText
  slug          String   @default(cuid())
  image         String?  @db.Text
  imageBlurhash String?  @db.LongText
  createdAt     DateTime @default(now())
  updatedAt     DateTime @updatedAt
  published     Boolean  @default(false)
  site          Site?    @relation(fields: [siteId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  siteId        String?

  @@unique([id, siteId], name: "post_site_constraint")
}

model LandingPageContent {
  id            String   @id @default(cuid())
  title         String?  @db.Text
  description   String?  @db.Text
  content       String?  @db.LongText
  slug          String   @default(cuid())
  image         String?  @db.Text
  imageBlurhash String?  @db.LongText
  createdAt     DateTime @default(now())
  updatedAt     DateTime @updatedAt
  site          Site?    @relation(fields: [siteId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  siteId        String?

  @@unique([id, siteId], name: "landing_page_constraint")
}

model Site {
  id            String   @id @default(cuid())
  name          String?
  description   String?  @db.Text
  logo          String?
  image         String?  @db.Text
  imageBlurhash String?  @db.LongText
  subdomain     String?  @unique
  customDomain  String?  @unique
  createdAt     DateTime @default(now())
  updatedAt     DateTime @updatedAt
  user          User?    @relation(fields: [userId], references: [id])
  userId        String?
  posts         Post[]

  LandingPageContent LandingPageContent[]
}

model Example {
  id            Int     @id @default(autoincrement())
  name          String?
  description   String? @db.Text
  domainCount   Int?
  url           String?
  image         String? @db.Text
  imageBlurhash String? @db.LongText
}

```

So, I have tried `npx prisma db push` and I just get the same error.

**Reply by nurul3101** [reply]
Is your `DATABASE_URL` variable also pointing to the connection string of `add-landingpage-table` branch?

**Reply by pantharshit00** [reply]
👋 Hello @christopheragnus, did you perhaps have a chance to check @nurul3101's recommendation? Let us know if you still have any questions!
 
Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by americo** [comment]
I had the same problem, but I managed to solve it.
The solution was to disable safe migrations on the planetscale database, as soon as I did that it worked.

But then I re-enabled safe migrations, don't forget to re-enable that.

**Reply by nurul3101** [reply]
Thanks for sharing the solution! 🙂 
This would definitely be helpful to users who might run into this issue later.

**Reply by carlos-ssh** [reply]
You saved me half day of my life! 
🙏 _IT WORKED FOR ME_ !

**Reply by jiwoniverse** [reply]
It's important to re-enable after disable! It worked for me too!

**Comment by sundaram2021** [comment]
I had the same issue and I found the solution for it , 
you just have to enable `Automatically copy migration data ` in the settings of planetscale 
and you are good to go

![image](https://github.com/prisma/prisma/assets/93595231/359e37d0-7d02-46e5-9f40-9d4f3409e47a)

After that you can run command `npx prisma db push` and it works fine

**Comment by Danubius1st** [comment]
I created a new pass for dev branch and set url accordingly.
```
datasource db {
  provider     = "mysql"
  url          = env("DATABASE_DEV_URL")
  // directUrl    = env("DIRECT_URL")
  relationMode = "prisma"
}

```
`npx prisma db push` works ok.
You can keep checked `Automatically copy migration data` and `Enable safe migrations`.

Doesn't work the example :
`DATABASE_URL="mysql://root@127.0.0.1:3309/prisma-playground"` in [documentation](https://planetscale.com/docs/prisma/automatic-prisma-migrations)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
